### PR TITLE
Update workshop notebooks from feedback

### LINKED ISSUE #406: Update annotated_images.show_sample() example
**Is your documentation update related to a problem? Please describe.**
Example in docs is wrong: https://mapreader.readthedocs.io/en/latest/User-guide/Classify/Train.html

**Describe the solution you'd like**
correct .show_image_labels() code snipped to .show_sample()

**Describe alternatives you've considered**

**Additional context**
discussed with Rosie during the training session

### LINKED ISSUE #408: Broken URL in the Workshop_AprilMay2024_part2.ipynb notebook
The URL to the `Classify` documentation webpage is broken in the `Create datasets and dataloaders` subsection.
This link https://mapreader.readthedocs.io/en/latest/User-guide/Classify.html#prepare-datasets-and-dataloaders
needs to be replaced by 
https://mapreader.readthedocs.io/en/latest/User-guide/Classify/Train.html

### LINKED ISSUE #409: Invalid syntax error on the calculate_add_metrics cell
There is a syntax error in the following cell: 

```python
#EXAMPLE
my_classifier.calculate_add_metrics(a
    y_true=my_classifier.orig_label,
    y_pred=my_classifier.pred_label,
    y_score=my_classifier.pred_conf,
    phase="test",
)
```
found in this page: https://mapreader.readthedocs.io/en/latest/User-guide/Classify/Train.html#testing 
The `a` must be removed after the parenthesis.

### LINKED ISSUE #410: Displaying scores per class
For those using multiclass classification, it would be interesting if you add an example in the documentation to display the scores per class. This can be very useful, especially for unbalanced datasets.
For instance, for displaying fscore on the test set:
```python
for label_id, label_name in annotated_images.labels_map.items():
    print(label_name, my_classifier.metrics['epoch_fscore_'+str(label_id)+'_test'])
```
I suggest adding this before the `Saving your work` section: https://mapreader.readthedocs.io/en/latest/User-guide/Classify/Train.html#saving-your-work

### LINKED ISSUE #411: Add the device argument for GPU support on the ClassifierContainer()
I would suggest adding the device argument in the ClassifierContainer() constructor in the documentation (and also in the Workshop_AprilMay2024.ipynb notebook): 
https://mapreader.readthedocs.io/en/latest/User-guide/Classify/Infer.html#initialize-classifiercontainer
for support for GPU: Nvidia chips (`cuda`) or Apple Mx chips (`mps`).

```python 
import torch

my_model = torch.load("./models/model_checkpoint_6.pkl")
labels_map = {0: "no_railspace", 1: "railspace"}
device = 'cuda' if torch.cuda.is_available() else 'mps' if torch.backends.mps.is_available() else 'cpu'

my_classifier = ClassifierContainer(my_model, labels_map, device=device)
```

For instance, running the inference example from the Workshop_AprilMay2024.ipynb notebook on a M1 Max Macbook laptop, the difference is 20 sec using GPU support against 9 min on CPU.

### LINKED ISSUE #420: review notebooks and provide suggestions for how to frame "your turn" sections

### PR DESCRIPTION
**Pull request by rwood-97**
### Summary

This addresses some of the TODOs in #417 .
Fixes #420 
Fixes #411 
Fixes #408 
Fixes #409 
Fixes #406 
Fixes #410 

### Describe your changes

Adds links to relevant documentation in the notebooks. 

TODO::

 - [x] Check each YOUR_TURN aligns with the code
 
### Checklist before assigning a reviewer (update as needed)

- [x] Self-review code
- [x] Ensure submission passes current tests
- [x] Add tests
- [x] Update relevant docs

### Reviewer checklist

Please add anything you want reviewers to specifically focus/comment on.

- [x] Everything looks ok?

### CODE REVIEW

**Comment by kmcdono2** [comment]
I've gone through the deep_solo, download, and the 2 data viz notebooks and added comments - not too much :) Will do parts 1 & 2 in the morning.
